File: algorithms/409. Longest Palindrome/solution.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        l = 0
        counter = collections.defaultdict(int)
        for c in s:
            if counter[c]:
                l += 2
                counter[c] = 0
            else:
                counter[c] = 1
        return l + (sum(counter.itervalues()) > 0)

# 95 / 95 test cases passed.
# Status: Accepted
# Runtime: 45 ms

File: algorithms/409. Longest Palindrome/solution.cc
class Solution {
public:
    int longestPalindrome(string s) {
        int l = 0;
        unordered_map<int, int> counter;
        for ( char c : s ) {
            if ( counter[c] ) {
                l += 2;
                --counter[c];
            } else {
                ++counter[c];
            }
        }
        for ( auto & p : counter ) {
            if ( p.second ) {
                l += 1;
                break;
            }
        }
        return l;
    }
};

// 95 / 95 test cases passed.
// Status: Accepted
// Runtime: 6 ms
